### PR TITLE
Balance sheet statements ARD update

### PR DESCRIPTION
**Pull request by mreis-ch**
Add method to StatementService to set agreed legal statement to false if they exist
Add calls to above method from the update method in SmallFullService
Update calls to method in the create and delete methods in ProfitAndLossService

BI-3184

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/api/accounts/service/impl/SmallFullService.java [hunk 1]
```diff
@@ -49,6 +49,9 @@
 
     private CompanyService companyService;
 
+    @Autowired
```
Comment (ch-tstrickland) [inline src/main/java/uk/gov/companieshouse/api/accounts/service/impl/SmallFullService.java:52]: Constructor is already autowired, which is how we should be doing it everywhere anyway, so this field ought to be not autowired AND be passed into the autowired constructor.

**Comment by ch-tstrickland** [comment]
Perhaps you forgot to check in the new unit tests?

File: src/main/java/uk/gov/companieshouse/api/accounts/service/impl/SmallFullService.java (hunk 1 above)
Comment (mreis-ch) [reply src/main/java/uk/gov/companieshouse/api/accounts/service/impl/SmallFullService.java:52]: I did initially put the StatementService in the constructor but on start up was getting a cyclic error which I couldn't quite sort. I will take another look

Comment (ch-tstrickland) [reply src/main/java/uk/gov/companieshouse/api/accounts/service/impl/SmallFullService.java:52]: I seem to remember Brad saying that this was a reason to do it this way - that it shows up this kind of issue.

**Pull request merged**